rust/agama-server/src/software/web.rs

```diff
@@ -463,3 +464,27 @@ async fn probe(State(state): State<SoftwareState<'_>>) -> Result<Json<()>, Error
     state.software.probe().await?;
     Ok(Json(()))
 }
+
+/// Updates the resolvables list with the given `id`.
+#[utoipa::path(
+    put,
+    path = "/resolvables/:id",
+    context_path = "/api/software",
+    responses(
+        (status = 200, description = "Read repositories data"),
+        (status = 400, description = "The D-Bus service could not perform the action
+")
+    )
+)]
+async fn set_resolvables(
+    State(state): State<SoftwareState<'_>>,
+    Path(id): Path<String>,
+    Json(params): Json<ResolvableParams>,
+) -> Result<Json<()>, Error> {
+    let names: Vec<_> = params.names.iter().map(|n| n.as_str()).collect();
```

Comment:
why type is needed here? I kind of expect it can infer that it is Vector from using collect.

Reply:
Yes, it is needed because the type cannot be inferred from the usage (it is used as a `[&str]` later). Bear in mind that `collect` can be used to build different kind of collections, even a `Result<T, E>`.